### PR TITLE
fixed sre issue related to beta v4 of JS

### PR DESCRIPTION
**Pull request by reeshika-h**
fixed vulnerability issue present in v4 beta branch of JS delivery sdk

### CODE REVIEW

File: src/lib/query.ts [hunk 1]
```diff
@@ -18,6 +18,24 @@ export class Query extends BaseQuery {
...
+  private isValidAlphanumeric(input: string): boolean {
+    const alphanumericRegex = /^[a-zA-Z0-9_.-]+$/;
+    return alphanumericRegex.test(input);
+  }
+  // Validate if input is a valid regex pattern
+  private isValidRegexPattern(input: string): boolean {
+    try {
+      new RegExp(input);
+      return true;
+    } catch (error) {
```
Comment (aman19K) [inline src/lib/query.ts:31]: @reeshika-h , [Minor] unused error object

File: src/lib/query.ts [hunk 2]
```diff
@@ -18,6 +18,24 @@ export class Query extends BaseQuery {
...
   }
+  // Validate if input is alphanumeric 
+  private isValidAlphanumeric(input: string): boolean {
+    const alphanumericRegex = /^[a-zA-Z0-9_.-]+$/;
+    return alphanumericRegex.test(input);
+  }
+  // Validate if input is a valid regex pattern
+  private isValidRegexPattern(input: string): boolean {
+    try {
+      new RegExp(input);
```
Comment (aman19K) [inline src/lib/query.ts:29]: @reeshika-h , [Minor] we can directly call RegExp(input) without the new keyword, by avoiding unnecessary object instantiation.

**Review by aman19K — CHANGES_REQUESTED** [review]

File: src/lib/query.ts (hunk 2 above)
Comment (reeshika-h) [reply src/lib/query.ts:29]: okay

**Review by aman19K — APPROVED** [review]

**Review by harshithad0703 — APPROVED** [review]
Approved

**Review by nadeem-cs — APPROVED** [review]
Approved.

**Pull request merged**